src/Services/Utils.php

```diff
@@ -40,24 +40,28 @@ public function __construct(ManagerRegistry $doctrine, TranslatorInterface $tran
         $this->doctrine = $doctrine;
     }
 
-    /**
-     * Hash a password according to the realm.
-     * Important note: It is very insecure and this is used only for the legacy sabre/dav implementation.
-     */
-    public function hashPassword(string $username, string $password): string
+    public function getRandomBytes($nbBytes = 32)
     {
-        return md5($username.':'.$this->authRealm.':'.$password);
+        $bytes = openssl_random_pseudo_bytes($nbBytes, $strong);
+        if (false !== $bytes && true === $strong) {
+            return $bytes;
+        }
+        else {
+            throw new \Exception("Unable to generate secure token from OpenSSL.");
+        }
+    }
+
+    public function generatePassword($length){
+        return substr(preg_replace("/[^a-zA-Z0-9]/", "", base64_encode($this->getRandomBytes($length+1))),0,$length);
     }
 
     public function createPasswordlessUserWithDefaultObjects(string $username, string $displayName, string $email)
     {
         $user = new User();
         $user->setUsername($username);
 
-        // Set the password to a random string (but hashed beforehand)
-        $randomBytes = random_bytes(256);
-        $hash = password_hash($randomBytes, PASSWORD_DEFAULT);
-        $user->setPassword($hash);
+        // Set the password to a secure randomly generated password
+        $user->setPassword($this->generatePassword(12));
```

Comment:
I don't understand how this fixes a "null error" as indicated in your PR description. Moreover, the `password_hash` function is the recommended, secure way of creating a password so I'm not sure of the rationale of implementing a custom function for that.

Reply:
This is the error I got when signing into OneCalendar with an IMAP user that had not been created yet with auto create on:
`app.ERROR: [500]: ValueError - Bcrypt password must not contain null character [{"file":"/usr/share/davis/src/Services/Utils.php","line":59,"function":"password_hash"},{"file":"/usr/share/davis/src/Services/IMAPAuth.php","line":62,"function":"createPasswordlessUserWithDefaultObjects","class":"App\\Services\\Utils","type":"->"},{"file":"/usr/share/davis/vendor/sabre/dav/lib/DAV/Auth/Backend/IMAP.php","line":80,"function":"imapOpen","class":"App\\Services\\IMAPAuth","type":"->"},{"file":"/usr/share/davis/vendor/sabre/dav/lib/DAV/Auth/Backend/AbstractBasic.php","line":103,"function":"validateUserPass","class":"Sabre\\DAV\\Auth\\Backend\\IMAP","type":"->"},{"file":"/usr/share/davis/vendor/sabre/dav/lib/DAV/Auth/Plugin.php","line":179,"function":"check","class":"Sabre\\DAV\\Auth\\Backend\\AbstractBasic","type":"->"},{"file":"/usr/share/davis/vendor/sabre/dav/lib/DAV/Auth/Plugin.php","line":135,"function":"check","class":"Sabre\\DAV\\Auth\\Plugin","type":"->"},{"file":"/usr/share/davis/vendor/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"beforeMethod","class":"Sabre\\DAV\\Auth\\Plugin","type":"->"},{"file":"/usr/share/davis/vendor/sabre/dav/lib/DAV/Server.php","line":456,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/usr/share/davis/vendor/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/usr/share/davis/src/Controller/DAVController.php","line":341,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/usr/share/davis/vendor/symfony/http-kernel/HttpKernel.php","line":163,"function":"dav","class":"App\\Controller\\DAVController","type":"->"},{"file":"/usr/share/davis/vendor/symfony/http-kernel/HttpKernel.php","line":75,"function":"handleRaw","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"->"},{"file":"/usr/share/davis/vendor/symfony/http-kernel/Kernel.php","line":202,"function":"handle","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"->"},{"file":"/usr/share/davis/public/index.php","line":35,"function":"handle","class":"Symfony\\Component\\HttpKernel\\Kernel","type":"->"}] []`

Reply:
Ok, I'll fix the error, it's a new [bcrypt behaviour](https://php.watch/versions/8.3/releases/8.3.5) in PHP

Reply:
FYI server is running Nginx 1.1.8 & PHP-FPM 8.2.22 on Ubuntu

Reply:
Done in https://github.com/tchapi/davis/commit/c4faa171e2cf42b3d7b18d31dc84085ef996a9de